test/alphabet/union_alphabet.cpp

```diff
@@ -44,34 +44,155 @@
 
 using namespace seqan3;
 
-TEST(union_alphabet_test, default_ctr)
+TEST(union_alphabet_test, default_constructor)
 {
     using alphabet_t = union_alphabet<dna4, gap>;
-    alphabet_t{};
+    constexpr alphabet_t letter1{};
+
+    EXPECT_EQ(letter1.value, 0);
+}
+
+TEST(union_alphabet_test, base_alphabet_constructor)
+{
+    using alphabet_t = union_alphabet<dna4, dna5, gap>;
+
+    constexpr alphabet_t letter0{dna4::A};
+    constexpr alphabet_t letter1{dna4::C};
+    constexpr alphabet_t letter2{dna4::G};
+    constexpr alphabet_t letter3{dna4::T};
+    constexpr alphabet_t letter4{dna5::A};
+    constexpr alphabet_t letter5{dna5::C};
+    constexpr alphabet_t letter6{dna5::G};
+    constexpr alphabet_t letter7{dna5::T};
+    constexpr alphabet_t letter8{dna5::N};
+    constexpr alphabet_t letter9{gap::GAP};
+
+    EXPECT_EQ(letter0.value, 0);
```

Comment:
For aggregate types it is exposed to aggregate initialization and structured bindings which is why we chose strictly typed enums for the value member so you don't write to it or read from it accidentally (e.g. via integrals or characters).
For non-aggregate POD class types the member isn't exposed at all, i.e. there is no way to know about it, unless you look at the implementation. We can discuss renaming it to `._value` if that makes you more happy, but it's not becoming part of the interface, it just doesn't make any sense!

Btw, it's what the STL (and every sensible library) does, too, or has to do. You can check it with `std::tuple` for example:
```c++
std::tuple<int, int> tup{7, 2};

std::cout << tup._M_head(tup) << '\n'; // prints 7
```
`_M_head()` is public, but it still implementation and not part of the interface.